src/NServiceBus.AmazonSQS/MessagePump.cs

```diff
@@ -438,35 +438,9 @@ async Task DeleteMessageAndBodyIfRequired(Message message, TransportMessage tran
                 return; // if another receiver fetches the data from S3
             }
 
-            if (transportMessage != null)
+            if (!string.IsNullOrEmpty(transportMessage?.S3BodyKey))
```

Comment:
should this be `IsNullOrWhiteSpace` just to be sure we're not emitting false positive logs info?

Reply:
Fine by me.